Fixed #12978 -- Added support for RSS feed stylesheets. (#18120)
django/django: docs/ref/contrib/syndication.txt

        ttl = 600  # Hard-coded Time To Live.

        # XSL STYLESHEET URL -- One of the following three is optional. The
        # framework looks for them in this order.

        def xsl_stylesheet_url(self, obj):
            """
            Takes the object returned by get_object() and returns the feed's

>             stylesheet URL (only XSl is supported)

Reviewer: ```suggestion
            stylesheet URL (only XSL is supported)
```

        ttl = 600  # Hard-coded Time To Live.

>         # STYLESHEET -- One of the following three is optional. The framework

Author: Agreed 👍🏻

        # The framework looks for them in this order.

        def stylesheets(self, obj):
            """
            Takes the object returned by get_object() and returns the feed's
            stylesheets (as URL strings or as Stylesheet instances).
            """

        def stylesheets(self):
            """
            Returns the feed's stylesheets (as URL strings or a Stylesheet
            instances).
            """

        # Hardcoded stylesheets

>         stylesheets = ["/stylesheet1.xsl", "stylesheet2.xls"]

Reviewer 2: ```suggestion
        stylesheets = ["/stylesheet1.xsl", "stylesheet2.xsl"]
```
Typo?

Author: Ah yes, good catch 😅

        ttl = 600  # Hard-coded Time To Live.

        # STYLESHEETS -- Optional. To set, provide one of the following three.
        # The framework looks for them in this order.

        def stylesheets(self, obj):
            """
            Takes the object returned by get_object() and returns the feed's
            stylesheets (an URL string or a Stylesheet instance).
            """

        def stylesheets(self):
            """
            Returns the feed's stylesheet (an URL string or a Stylesheet
            instance).
            """

        # Hardcoded stylesheets (can be a single stylesheet or a list)
        stylesheets = "/link/to/stylesheet.xsl"
        stylesheets = feedgenerator.Stylesheet("/link/to/stylesheet.xsl")
        stylesheets = ["/stylesheet1.xsl", "stylesheet2.xls"]
        stylesheets = [
            feedgenerator.Stylesheet("/stylesheet1.xsl"),
            feedgenerator.Stylesheet("/stylesheet2.xsl"),

>         ]

Author: It was a hard choice, but I went with the 3rd example 😁

    * ``feed_copyright``
    * ``feed_guid``
    * ``ttl``
    * ``stylesheet``

    Any extra keyword arguments you pass to ``__init__`` will be stored in

>     ``self.feed`` for use with `custom feed generators`_.

Reviewer 3: Further down here, we have the same "All parameters should be strings..." note where you might want to add "The ``stylesheet`` argument can be a string or an instance of :class:`Stylesheet`."

Author: Good catch, I've reused the same wording here as in the other file.

    * ``feed_copyright``
    * ``feed_guid``
    * ``ttl``
    * ``stylesheets``

    Any extra keyword arguments you pass to ``__init__`` will be stored in
    ``self.feed`` for use with `custom feed generators`_.

    All parameters should be strings, except for two:

    * ``categories`` should be a sequence of strings.
    * ``stylesheets`` can be a single string, an instance of
      :class:`~django.utils.feedgenerator.Stylesheet`, or a list of either
      strings or ``Stylesheet`` instances.

    Beware that some control characters are
    `not allowed <https://www.w3.org/International/questions/qa-controls>`_ in
    XML documents. If your content has some of them, you might encounter a

>     :exc:`ValueError` when producing the feed.

Author: Ah yes, good point

There's a lot more work to be done for a complete custom feed class, but the
above example should demonstrate the basic idea.

Feed stylesheets
----------------

.. versionadded:: 5.1

>     The ability to specify ``xsl_stylesheet_url`` was added.
> 

Reviewer 4: This is unnecessary
```suggestion
```

> an ``xsl_stylesheet_url`` argument to the feed's constructor.

Reviewer: Just highlighting this, it may well be fine -- `CTRL-F` suggests this is the only place in the document where we refer to the `__init__` as a _constructor_, I can't tell at a glance what the rest of the document uses as nomenclature 🤷

Author: Thanks for pointing this out, I had actually gotten myself mixed up between `django.contrib.syndication.views.Feed` and `django.utils.feedgenerator.SyndicationFeed` and the documentation I'd written didn't actually work.

`SyndicationFeed` is the one that takes arguments in its constructor (this behavior worked and was tested) but `Feed` is customized by settings arguments on the class (this behavior had not been tested and did not work).

I've fixed the missing code and updated the docs 👍🏻

You can add styling to your RSS feed using an `XSLT stylesheet`_ by setting the
``xsl_stylesheet_url`` attribute on the feed class.

This can be a hardcoded URL::

    from django.contrib.syndication.views import Feed

    class FeedWithHardcodedStylesheet(Feed):
        ...  # author, etc.
        xsl_stylesheet_url = "https://example.com/rss_stylesheet.xslt"

You can also use Django's static files system::

    from django.contrib.syndication.views import Feed
    from django.templatetags.static import static

    class FeedWithStaticFileStylesheet(Feed):
        ...  # author, etc.
        xsl_stylesheet_url = static("rss_styles.xslt")

> 

Reviewer 4: ```suggestion
```

    from django.urls import reverse

    class FeedWithStylesheetView(Feed):
        ...  # author, etc.

>         xsl_stylesheet_url = reverse("your-custom-view-name")

Reviewer: 🤔  Does this example need to use `reverse_lazy` by convention, or is continuing to use `reverse`[^1] going to be OK here?

[^1]: As earlier in the document, but at method evaluation level, not at module level.

.. _xslt stylesheet: https://developer.mozilla.org/en-US/docs/Web/XSLT/Transforming_XML_with_XSLT

.. versionadded:: 5.1

> The ability to specify ``xsl_stylesheet_url`` was added.

Reviewer 4: This should go to the top:
```
Feed stylesheets
----------------

.. versionadded:: 5.1

```

There's a lot more work to be done for a complete custom feed class, but the
above example should demonstrate the basic idea.

Feed stylesheets
----------------

.. versionadded:: 5.1

> You can add styling to your RSS feed using a stylesheet (typically in the XSLT_
> or CSS formats) by setting the ``stylesheets`` attribute on the feed class.

Reviewer 3: ```suggestion
If you wish to have your RSS feed render nicely in a browser, you will need to
provide styling information for the XML file, typically in XSLT_ or CSS formats.

You can add this to your RSS feed by setting the ``stylesheets`` attribute on the
feed class.
```
Maybe?

There's a lot more work to be done for a complete custom feed class, but the
above example should demonstrate the basic idea.

Feed stylesheets
----------------

.. versionadded:: 5.1

You can add styling to your RSS feed using a stylesheet (typically in the XSLT_
or CSS formats) by setting the ``stylesheets`` attribute on the feed class.

> This can be a hardcoded URL::

Author: I don't think that's necessary. The `::` introduces a code block with the default language (which should be Python). On my local build the code examples are correctly highlighted as Python, and the rest of the document uses the same syntax (except for a few examples which have an explicit `code-block:: django+html`)
